File: 53-maximum-subarray/53-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxi=-10000
        pre=0
        for i in range(len(nums)):
            pre+=nums[i]
            if pre>maxi:
                maxi=pre
            if pre<0:
                pre=0
        return maxi

File: 53-maximum-subarray/53-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxsum=-99999;
        int check=0;
        for(int i=0;i<nums.length;i++){
            check+=nums[i];
            if (check > maxsum){
                maxsum=check;
            }
            if (check<0){
                check=0;
            }
        }
        return maxsum;
            
        
    }
}